File: backend/bff-server/src/main/java/arile/toy/stocksystem/bffserver/stockinfo/dto/StockInfo.java
package arile.toy.stocksystem.bffserver.stockinfo.dto;

public record StockInfo(
        String marketCap,
        String listedShares,

        String parValue,
        String tradingUnit,

        String foreignLimit,
        String foreignOwned,
        String foreignRate,

        String opinion,
        String targetPrice,
        String consensusDate,

        String high52,
        String low52,

        String per,
        String eps,

        String estimatedPer,
        String estimatedEps,

        String pbr,
        String bps,

        String dividendYield,

        String sameIndustryPer,
        String sameIndustryRate
) {
}

File: frontend/src/types/stockInfo.ts
export interface StockInfoResponse {
    marketCap: string
    marketCapRank: string
    listedShares: string

    parValue: string
    tradingUnit: string

    foreignLimit: string
    foreignOwned: string
    foreignRate: string

    opinion: string
    targetPrice: string

    high52: string
    low52: string

    per: string
    eps: string

    estimatedPer: string
    estimatedEps: string

    pbr: string
    bps: string

    dividendYield: string

    sameIndustryPer: string
    sameIndustryRate: string
}
